configure_xxx.go file gets overwritten everytime.

## Problem statement
Even though there is a message in such file saying
`// This file is safe to edit. Once it exists it will not be overwritten`
the file gets always rewritten by calling `swagger generate server...`

## Swagger specification
any

## Steps to reproduce
1. Execute 'swagger generate server -f swagger.json'
2. Modify the file generated which starts with 'configure_' and sits in restapi folder
3. Execute 'swagger generate server -f swagger.json' again

It seems like the problem is in ToFileName function from github.com\go-openapi\swag\util.go
I extended the test for this function with case: `{"FileWithExtension.go", "file_with_extension.go"}`

And it failed with this message:
```{r, engine='bash', count_lines}
--- FAIL: TestToFileName (0.01s)
        Error Trace:    util_test.go:133
        Error:          Not equal:
                        expected: "file_with_extension.go"
                        received: "file_with_extension_go"
```

--- a/circle.yml
+++ b/circle.yml
@@ -52,7 +52,8 @@ deployment:
     branch: master
     owner: go-swagger
     commands:
-      - mkdir -p dist/linux/amd64/usr/bin
+      - mkdir -p dist
+      - docker run --rm -it -v `pwd`:/go/src/github.com/go-swagger/go-swagger -w /go/src/github.com/go-swagger/go-swagger golang:1.7-alpine go build -o ./dist/swagger-musl  -a -tags netgo -installsuffix netgo ./cmd/swagger
       - docker login -u $API_USERNAME -p $QUAY_PASS -e $API_EMAIL https://quay.io
       - docker build -t quay.io/goswagger/swagger:dev .
       - docker push quay.io/goswagger/swagger

--- a/generator/shared.go
+++ b/generator/shared.go
@@ -392,7 +392,7 @@ func (g *GenOpts) location(t *TemplateOpts, data interface{}) (string, string, e
 	if fNameTpl.Execute(&fNameBuf, d); err != nil {
 		return "", "", err
 	}
-	return pthBuf.String(), fNameBuf.String(), nil
+	return pthBuf.String(), fileName(fNameBuf.String()), nil
 }
 
 func (g *GenOpts) render(t *TemplateOpts, data interface{}) ([]byte, error) {
@@ -460,7 +460,7 @@ func (g *GenOpts) write(t *TemplateOpts, data interface{}) error {
 		}
 	}
 
-	return ioutil.WriteFile(filepath.Join(dir, fileName(fname)), formatted, 0644)
+	return ioutil.WriteFile(filepath.Join(dir, fname), formatted, 0644)
 }
 
 func fileName(in string) string {
@@ -558,8 +558,7 @@ func loadSpec(specFile string) (string, *loads.Document, error) {
 }
 
 func fileExists(target, name string) bool {
-	ffn := swag.ToFileName(name)
-	_, err := os.Stat(filepath.Join(target, ffn))
+	_, err := os.Stat(filepath.Join(target, name))
 	return !os.IsNotExist(err)
 }
 

